src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/ArrayOperations.cs

```diff
@@ -47,12 +47,10 @@ internal NewArrayInstruction(Type elementType)
         public override int Run(InterpretedFrame frame)
         {
             int length = ConvertHelper.ToInt32NoNull(frame.Pop());
-            if (length < 0)
-            {
-                // to make behavior aligned with array creation emitted by C# compiler
-                throw new OverflowException();
-            }
-            frame.Push(Array.CreateInstance(_elementType, length));
+            // To make behavior aligned with array creation emitted by C# compiler if length is less than
+            // zero we try to use it to create an array, which will throw an OverflowException with the
+            // correct localized error message.
+            frame.Push(length < 0 ? new int[length] : Array.CreateInstance(_elementType, length));
```

Comment:
In the spirit of https://github.com/dotnet/corefx/issues/21825, it would be nice if the message contained the bad length.